pytorch_lightning/trainer/trainer.py

```diff
@@ -94,15 +94,63 @@ def __init__(
         Customize every aspect of training via flags
 
         Args:
-            logger (:class:`.Logger`): Logger for experiment tracking
-            example:
-                >>> asd
-            checkpoint_callback (:class:`CheckpointCallback`): Callback for checkpointing
+            logger (:class:`.Logger`): Logger for experiment tracking.
+                Example::
+                    from pytorch_lightning.logging import TensorBoardLogger
+
+                    # default logger used by trainer
+                    logger = TensorBoardLogger(
+                        save_dir=os.getcwd(),
+                        version=self.slurm_job_id,
+                        name='lightning_logs'
+                    )
+
+                    Trainer(logger=logger)
+            checkpoint_callback (:class:`CheckpointCallback`): Callback for checkpointing.
+                Example::
+                    from pytorch_lightning.callbacks import ModelCheckpoint 
+
+                    # default used by the Trainer
+                    checkpoint_callback = ModelCheckpoint(
+                        filepath=os.getcwd(),
+                        save_best_only=True,
+                        verbose=True,
+                        monitor='val_loss',
+                        mode='min',
+                        prefix=''
+                    )
+
+                    trainer = Trainer(checkpoint_callback=checkpoint_callback)
             early_stop_callback (:class:`.EarlyStopping`): Callback for early stopping
-            default_save_path (str): Default path for logs+weights if no logger/ckpt_callback passed
-            gradient_clip_val (int): 0 means don't clip.
-            gradient_clip (int): 0 means don't clip. Deprecated.
-            process_position (int): shown in the tqdm bar
+                Example::
```

Comment:
just a reminder, it still will be on the same line as "early_stop_callback (:class:`.EarlyStopping`): Callback for early stopping Example"

Reply:
i would wait a bit until this is more done hahaha. this will continue to change :)